internal/controller/promotionstrategy_controller_test.go

```diff
@@ -312,30 +312,30 @@ var _ = Describe("PromotionStrategy Controller", func() {
 				g.Expect(promotionStrategy.Status.Environments).To(HaveLen(3))
 
 				// By("Checking that the ChangeTransferPolicy for development has shas that are not empty, meaning we have reconciled the resource")
-				// Active dry sha should be empty because hydrator.metadata is not present
+				// Active dry sha should be the same as proposed dry sha because we should have merged the branch with hydrator.metadata.
```

Comment:
idk why this test was passing in the first place... iiuc if hydrator.metadata is missing on the active branch, we should treat it the same as an outdated SHA and merge the proposed branch in.

I've changed the tests to assume we go ahead and merge the proposed branch (with hydrator.metadata) into the active branch.